src/app/components/ResourceStatusLabel/ResourceStatusLabel.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { forwardRef } from "react";
+import {
+  ResourceStatus,
+  ResourceStatusDelayed,
+} from "@app/components/ResourceStatusLabel/types";
+import {
+  Alert,
+  Button,
+  Flex,
+  FlexItem,
+  HelperText,
+  HelperTextItem,
+  Spinner,
+  Split,
+  SplitItem,
+} from "@patternfly/react-core";
+import {
+  CheckCircleIcon,
+  ExclamationTriangleIcon,
+} from "@patternfly/react-icons";
+import { useTranslation } from "react-i18next";
+import "./ResourceStatusLabel.css";
+
+export interface ResourceStatusLabelProps {
+  status: ResourceStatus;
+  creationDelayed?: ResourceStatusDelayed;
+}
+
+export const ResourceStatusLabel = forwardRef<
+  HTMLButtonElement,
+  ResourceStatusLabelProps
```

Comment:
This is a generic status label component for resources. 
Its status prop accepts only the "UI statuses", meaning the ones displayed in the label text.